sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -245,6 +246,56 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
       }.toArray
       DeleteFromTableExec(r.table.asDeletable, filters) :: Nil
 
+    case UpdateTable(maybeRelation, attrs, values, condition) =>
+      val nested = attrs.asInstanceOf[Seq[Any]].filterNot(_.isInstanceOf[AttributeReference])
+      if (nested.nonEmpty) {
+        throw new AnalysisException(s"Update only support non-nested fields. Nested: $nested")
+      }
+
+      val (relation, attrsNames, newValues, newCond) = maybeRelation match {
+        case d: DataSourceV2Relation =>
+          (d, attrs.map(_.name), values, condition)
+        case Project(aliasList: Seq[Alias], r: DataSourceV2Relation) =>
```

Comment:
This seems like too much code. Can you check with other databases and see if they support column alias as well? If most of them don't support it, we can throw exception in the parser if column alias is used.

Reply:
I've checked 

> @xianyinxin, can you explain the required semantics for your proposed API?
> 
> ```
> void updateWhere(Map<String, Expression> sets, Filter[] filters);
> ```
> 
> It isn't clear what the requirement for a source would be.
> 
> In addition, `Expression` is internal to catalyst and should be removed from the API.

Thank you @rdblue . Here `Expression` is the public datasource expression `org.apache.spark.sql.catalog.v2.expressions.Expression`. Datasource needs to know the value that the field be updated to, so here the key of `sets` specifies the field to be updated, and the value of `sets` is the updated value. The value can be an expression, not just a literal, IMHO, like the case `UPDATE tbl SET a=a+1 WHERE ...

Reply:
I checked mysql, postgresql, hive and impala. All of these do not support column alias, and most does not support table alias. I believe it is not a standard syntax and I'd prefer to remove the column alias and only keep the table alias.

Reply:
Let's only keep table alias. We can change the parser, replace `tableAlias` with `(AS? strictIdentifier)?`.

Reply:
OK for me, will update it.

Reply:
> Let's only keep table alias. We can change the parser, replace `tableAlias` with `(AS? strictIdentifier)?`.

Emm, they are two opts: 1. replace `tableAlias` with `(AS? strictIdentifier)?`; 2. keep `tableAlias` and throw `ParseException` when we checked it contains columns alias. The 1st opt will just throw a ParseException with meaningless hint, but the later we can define the error msg "`Columns aliases is not allowed in UPDATE.`". I think the latter is better?

Reply:
Yea the latter one provides better error message.